src/components/Advertiser/CampaignCard.vue

```diff
@@ -55,23 +56,31 @@ function goToUrl() {
   router.push('/campaign/' + props.advertise.id)
 }
 
-onMounted(async () => {
+onMounted(() => {
   articleRef.value.focus()
+
   const observer = new IntersectionObserver((entries) => {
     entries.forEach((entry) => {
-      const intersecting = entry.isIntersecting
-      if (intersecting && props.advertise?.status === 'Active') {
-        impressionsStore.addImpression('advertises', props.advertise?.id).catch((error) => errorStore.throwError(error))
+      if (entry.isIntersecting && props.advertise?.status === 'Active') {
+        const adId = props.advertise.id
+        // Increment count for each specific adId
+        impressions[adId] = (impressions[adId] || 0) + 1
+        LocalStorage.set(adId, impressions[adId])
```

Comment:
Nice catch. Forgot to remove this as I noticed there is no actual need to keep it in local storage for this implementation as I intendet at the begining, because it is stored in memory actually...